tratto/src/main/java/star/tratto/util/FileUtils.java

```diff
@@ -0,0 +1,124 @@
+package star.tratto.util;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import star.tratto.identifiers.FileFormat;
+import star.tratto.identifiers.FileName;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+/**
+ * I/O utilities: creating files, writing to files, retrieving all Java files, etc.
+ */
+public class FileUtils {
+    /**
+     * Combines given information into a file path.
+     *
+     * @param dirPath a base root path
+     * @param fileName the file name
+     * @param fileFormat the type of file (e.g. ".txt", ".json")
+     * @param projectName the corresponding project being analyzed
+     * @return a file path which combines the given information into
+     * format:
+     *  "[dirPath]/[projectName]/[fileName].[fileFormat.getExtension()]"
+     */
+    public static Path getPath(
+            String dirPath,
+            String fileName,
+            FileFormat fileFormat,
+            String projectName
+    ) {
+        return Paths.get(dirPath, projectName, fileName + fileFormat.getExtension());
+    }
+
+    /**
+     * Creates an empty file at a given path. Creates parent directories if
+     * necessary. If the file already exists, then this method does nothing.
+     *
+     * @param path a file
+     * @throws IOException if an error occurs while creating the parent
+     * directories or new file
+     */
+    public static void createFile(Path path) throws IOException {
```

Comment:
The methods in this file will be easier for clients to use if they throw Error instead of IOException, because then clients won't need to write a `catch` clause.  If you expect clients to locally handle the IOException, keep it as is.  If you expect no clients to locally handle the IOException, then change it to an Error here.